comp/core/workloadmeta/collectors/internal/remote/processcollector/process_collector.go

```diff
@@ -152,17 +156,56 @@ func (s *streamHandler) NewClient(cc grpc.ClientConnInterface) remote.GrpcClient
 	return &client{cl: pbgo.NewProcessEntityStreamClient(cc), parentCollector: s}
 }
 
-func (s *streamHandler) HandleResponse(resp interface{}) ([]workloadmeta.CollectorEvent, error) {
+// populateMissingContainerID populates any missing containerID field in the entities of Set events if it is possible to get the
+// container id from the shared container provider
+func (s *streamHandler) populateMissingContainerID(collectorEvents []workloadmeta.CollectorEvent, store workloadmeta.Component) {
```

Comment:
small nits. I would revert the function option and to support variadic `collectorEvents`

```suggestion
func (s *streamHandler) populateMissingContainerID(store workloadmeta.Component, collectorEvents workloadmeta.CollectorEvent...) {
```

Reply:
another small nits: I would define the function `populateMissingContainerID()` after `HandleResponse()` because `HandleResponse()` is public